minard/burst.py

```diff
@@ -47,3 +47,93 @@ def burst_run_detail(run_number, subrun, sub):
         files = "%i_%i_%i" % (run_number,subrun,sub)
 
     return result, files
+
+def load_bursts_search(search, start, end, offset, limit):
+    """
+    Returns a dictionary with the burst runs loaded from couchdb.
+    The returned dictionary is given by one of the search conditions on the page:
+    either by run, date or GTID which all use corresponding couchdb views.
+    """
+    server = couchdb.Server("http://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    db = server["burst"]
+
+    i = 0 #the counter keeps the ordering correct
+    results = {}
+    skip = offset
+
+    if (search == "run"):
+
+        startkey = [int(start), 0, {}]
+        endkey = [int(end), {}]
+
+        view = '_design/burst/_view/burst_by_run'
+
+    elif (search == "date"):
+
+        Syear = start[0:4]
+        Smonth = start[5:7]
+        Sday = start[8:10]
+        Eyear = end[0:4]
+        Emonth = end[5:7]
+        Eday = end[8:10]
+        if Smonth[0] == "0": Smonth = Smonth[1]
+        if Emonth[0] == "0": Emonth = Emonth[1]
+        if Sday[0] == "0": Sday = Sday[1]
+        if Eday[0] == "0": Eday = Eday[1]
+        startkey = [int(Syear), int(Smonth), int(Sday)]
+        endkey = [int(Eyear), int(Emonth), int(Eday)]
+
+        view = '_design/burst/_view/burst_by_date'
+
+    elif (search == "gtid"):
+
+        view = '_design/burst/_view/burst_by_GTID'
+
+    if ((search == "run") or (search == "date")):
```

Comment:
removed the parentheses.
If you don't mind I would like to keep the function in the current form so that I don't have to triplicate parts of the code. It's the question of simple to read vs redundant and duplicate code. Can change if you really don't like this.